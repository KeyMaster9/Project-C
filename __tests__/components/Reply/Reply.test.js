import React from 'react';
import renderer from 'react-test-renderer';

beforeEach(() => {
    jest.restoreAllMocks();
});

[2, 3].forEach(num => {
    it('should render a TravelAgent Component', () => {
        jest.mock('../../../src/components/Reply/Responses/Buttons.jsx', () => (props) => <span>{JSON.stringify(props)}</span>);
        jest.mock('../../../src/components/Reply/Responses/Select.jsx', () => (props) => <span>{JSON.stringify(props)}</span>);
        jest.mock('../../../src/components/Reply/Send/Send.jsx', () => (props) => <span>{JSON.stringify(props)}</span>)

        const mockOnSend = () => 'mock onSend';

        const mockQuestion = {
            property: 'test',
            question: 'this is the test question',
            responses: [...Array(num)].map((x, i) => `${i + 1}`),
        }

        const { default: Reply } = require('../../../src/components/Reply/Reply.jsx');
        const component = renderer.create(<Reply question={mockQuestion} onSend={mockOnSend} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})

