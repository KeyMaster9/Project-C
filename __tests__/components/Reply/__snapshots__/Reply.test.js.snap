// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a TravelAgent Component 1`] = `
<div
  className="Reply"
>
  <span>
    {"options":["1","2"]}
  </span>
  <span>
    {"isDisabled":true}
  </span>
</div>
`;

exports[`should render a TravelAgent Component 2`] = `
<div
  className="Reply"
>
  <span>
    {"options":["1","2","3"]}
  </span>
  <span>
    {"isDisabled":true}
  </span>
</div>
`;
